src/DependencyResolver/Package.php

```diff
@@ -31,9 +32,9 @@
 final class Package
 {
     /**
-     * @param list<ConfigureOption>       $configureOptions
-     * @param list<OperatingSystemFamily> $compatibleOsFamilies
-     * @param list<OperatingSystemFamily> $incompatibleOsFamilies
+     * @param list<ConfigureOption>                      $configureOptions
+     * @param non-empty-list<OperatingSystemFamily>|null $compatibleOsFamilies
```

Comment:
The key change here is that the OS families can either be `null` (explicitly not specified in `php-ext`), or a non-empty-list; I think this helps express intent better, otherwise, maybe someone could try:

```
  "os-families": []
```

which does not make sense